### PR TITLE
Adding task guides to resources

### PR DESCRIPTION
**Pull request by MKhalusova**
Previously, we have added links to compatible models to task guides. This PR enables navigation in the opposite direction and adds links to relevant task guides (based on model mapping) to the list of resources in model docs. Those who land on the model docs should now be able to find relevant task guides quicker. The links are added to the list of resources (along with previously listed notebooks, blog posts, scripts, etc.)

### CODE REVIEW

File: docs/source/en/model_doc/opt.mdx [hunk 1]
```diff
@@ -29,6 +28,35 @@ Tips:
...
+- [Causal language modeling](https://huggingface.co/course/en/chapter7/6?fw=pt#training-a-causal-language-model-from-scratch) chapter of the 🤗 Hugging Face Course.
+- [`OPTForCausalLM`] is supported by this [causal language modeling example script](https://github.com/huggingface/transformers/tree/main/examples/pytorch/language-modeling#gpt-2gpt-and-causal-language-modeling) and [notebook](https://colab.research.google.com/github/huggingface/notebooks/blob/main/examples/language_modeling.ipynb).
+- [`TFOPTForCausalLM`] is supported by this [causal language modeling example script](https://github.com/huggingface/transformers/tree/main/examples/tensorflow/language-modeling#run_clmpy) and [notebook](https://colab.research.google.com/github/huggingface/notebooks/blob/main/examples/language_modeling-tf.ipynb).
+- [`FlaxOPTForCausalLM`] is supported by this [causal language modeling example script](https://github.com/huggingface/transformers/tree/main/examples/flax/language-modeling#causal-language-modeling).
+
+<PipelineTag pipeline="text-classification" />
+
+- [Token classification](https://huggingface.co/course/chapter7/2?fw=pt) chapter of the 🤗 Hugging Face Course.
```
Comment (sgugger) [inline docs/source/en/model_doc/opt.mdx:48]: Should be Sequence classification and the corresponding chapter in the course.

File: docs/source/en/model_doc/vision-encoder-decoder.mdx [hunk 2]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (sgugger) [inline docs/source/en/model_doc/vision-encoder-decoder.mdx:1]: No need to add this here.
```suggestion
```

**Review by sgugger — APPROVED** [review]
Impressive work, thanks for adding all of those!

File: docs/source/en/model_doc/opt.mdx (hunk 1 above)
Comment (MKhalusova) [reply docs/source/en/model_doc/opt.mdx]: Fixed in a5bb3a015fddcdc70e6114da7ebed905deb222e7

**Pull request merged**